docs/ingester-handover.md

```diff
@@ -0,0 +1,44 @@
+# Ingester Hand-over
+
+The [ingester](architecture.md#ingester) holds several hours of sample
+data in memory. When we want to shut down an ingester, either for
+software version update or to drain a node for maintenance, this data
+must not be discarded.
+
+Each ingester goes through different states in its lifecycle. When
+working normally, the state is `ACTIVE`.
+
+On start-up, an ingester first goes into state `PENDING`. After a
+short time, if nothing happens, it adds itself to the ring and goes
+into state ACTIVE.
```

Comment:
Tuneable with `-ingester.join-after`

Reply:
If you add that property can you tell under which circumstances you may want to tune it?
Why wouldn't I like to have a new ingester to immediately join the ring (i. e. setting `-ingester.join-after` to 0). Would that introduce the risk the ingester is considered as additional ingester (rather than the new ingester which is supposed to receive the data from the terminating ingester?) or why does that property even exist?

I am missing some more context to understand what is happening there.

Reply:
The leaving ingester probes periodically for a joining one - need to allow time for this to happen.  I guess it's tuneable because we don't trust our initial guesses.

Personally I don't want to litter the main narrative with all the tuning options - maybe add a pointer to their description in `arguments.md` ?

Reply:
I think linking to `arguments.md` makes a lot of sense for the detailed parameters.